node/impl/full/eth.go

```diff
@@ -804,12 +805,137 @@ func (a *EthModule) EthEstimateGas(ctx context.Context, tx ethtypes.EthCall) (et
 	// gas estimation actually run.
 	msg.GasLimit = 0
 
-	msg, err = a.GasAPI.GasEstimateMessageGas(ctx, msg, nil, types.EmptyTSK)
+	ts := a.Chain.GetHeaviestTipSet()
+	msg, err = a.GasAPI.GasEstimateMessageGas(ctx, msg, nil, ts.Key())
 	if err != nil {
 		return ethtypes.EthUint64(0), err
```

Comment:
Here, and in other places, can we please chain errors? It makes debugging much easier.
```suggestion
		return ethtypes.EthUint64(0), xerrors.Errorf("failed to estimate gas: %w", err)
```